### PR TITLE
Friendly developer onboarding

### PR DESCRIPTION
**Pull request by yannvanhalewyn**
So it actually took me quite some time to go through the source to figure out how to start everything up correctly. I never use Datomic (their onboarding was a bit lacking too sadly), or mount. Bit since I did all this work figuring it out I documented it and added some basic and widely used tools around them. I've added some notes to some commits as to why for each step if you're interested to go through them.

### CODE REVIEW

**Comment by yannvanhalewyn** [comment]
cc @awkay

**Comment by yannvanhalewyn** [comment]
Also I added a few emacs related things. I'm comfortable doing this because I know it's a widely used editor and the things I've added should improve everyone's experience. I don't think this will interfere with your setup, but I can gitignore it if you prefer.

File: src/example/com/example/components/datomic.clj [hunk 1]
```diff
@@ -21,11 +21,19 @@
                     database "?user=" user "&password=" password)]
     url))
 
+(defn presence
+  "Returns s if s is not blank. Takes an optional default."
+  [s & [default]]
+  (if (str/blank? s)
+    default
+    s))
```
Comment (yannvanhalewyn) [inline src/example/com/example/components/datomic.clj]: You know it's gonna be fun when my first actual code commit to a rapid application framework is coercion

**Comment by awkay** [comment]
So, I know this is going to be a point of contention for people, but code formatting will be IntelliJ's with the settings I use, and I won't accept a PR that makes formatting changes unless it is really an error on my part (e.g. leaving a dangling paren). 

I use the following settings in IntelliJ:
- 2 space indent
- On general tab: Default to Only Indent

You can keep from committing accidental ws changes with this git alias in global .gitconfig:

```
[alias]
  addnw=!sh -c 'git diff -U0 -w --no-color "$@" | git apply --cached --ignore-whitespace --unidiff-zero -'
```

Then use:

```
$ git addnw
$ git checkout -- .
```

which will add the non-whitespace changes to the index, and then throw away the ws changes.

Be nice to add that to a contributor readme.

File: src/example/com/example/components/datomic.clj [hunk 2]
```diff
@@ -1,57 +1,77 @@
...
-    [com.fulcrologic.guardrails.core :refer [>defn =>]]
-    [com.example.schema :refer [prior-schema latest-schema]]
-    [taoensso.timbre :as log]
-    [datomic.api :as d]
-    [datomock.core :refer [mock-conn]]
-    [mount.core :refer [defstate]]
-    [taoensso.timbre :as log]
-    [com.example.components.config :refer [config]]
-    [clojure.string :as str]))
+   [com.fulcrologic.rad.database-adapters.db-adapter :as dba]
```
Comment (awkay) [inline src/example/com/example/components/datomic.clj]: 2 spaces of indent...sorry, this isn't something I want in any PR...not flip flop on indents.  In general don't commit ws changes, and fix your indentation to match the project: indentation is 2 spaces for each level of nesting.

**Comment by awkay** [comment]
I'm not going to pick on this too much, given that the entire source base is experimental, and there isn't much point nit-picking stuff that is likely to get thrown away or rewritten anyhow.  It's good that you got through it and understand it all...so I'm fine with merging this if you're sure it still works.

**Comment by awkay** [comment]
Please do squash and merge to make this one commit, though.

File: src/example/com/example/components/datomic.clj (hunk 2 above)
Comment (yannvanhalewyn) [reply src/example/com/example/components/datomic.clj]: No worries, I see things the same way. The thing is I wanted to mimic your IntelliJ indentation setup, but I found different styles in different files (or so I thought) so I picked one. I might've missed something. I'll change it to 2 spaces 😉

**Comment by yannvanhalewyn** [comment]
@awkay 

> You can keep from committing accidental ws changes with this git alias in global .gitconfig:

TIL, didnt' know that

> Please do squash and merge to make this one commit, though.

This I don't understand. I rewrote the history to make sense in steps, and it would become a merge commit on master. Why should these commits be squashed?

I'll change the indentation.

**Comment by awkay** [comment]
I'm not tied to the squash...I typically squash mine just to make history shorter because I do a lot of little fiddly commits at times, and it just makes it easier to read the progression...but if your commits are well-planned, I don't really care.

**Pull request merged**